### PR TITLE
chore: Marking database as fresh whenever a new database is created in Spanner tests.

### PR DESCRIPTION
**Pull request by anuragsrivstv**
This PR contains changes to mark database as fresh whenever a new database is created in Spanner tests.

Resolves issue #9204

### CODE REVIEW

**Comment by anuragsrivstv** [comment]
I'll update the commit message to make it same as PR description.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs
Comment (Rishabh-V) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs:115]: I haven't looked in detail, but it appears that this will now always be true, irrespective of if database is created or not. Am I missing something?

Comment (Rishabh-V) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs:115]: And what will happen if database already exists? Before Fresh is true, exception will be thrown, based on current implementation of GSQL  `CreateDatabase`

Comment (Rishabh-V) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs:115]: Possibly, change `protected abstract void CreateDatabase();` to return bool. And return `True` if fresh and `False` if it already exists.

Comment (anuragsrivstv) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs]: Thanks for feedback. Rectified the code.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs
Comment (Rishabh-V) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs]: This is a `bool` with a default value of `false`. Do we need to set it explicitly?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs
Comment (Rishabh-V) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs:114]: This message isn't correct now. If database is created fresh, it should be "created message" else "using existing database" message,

Comment (Rishabh-V) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs:114]: Also, I see we provide message in the `TryCreateDatabase` methods. We can probably optimize that.

Comment (anuragsrivstv) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs]: corrected log.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabasePostgreSql.cs
Comment (Rishabh-V) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabasePostgreSql.cs:92]: We have the logger message here and we have removed it from the other class. Possibly keep the create database message in the derived classes as they are different and in case of reusing the database (i.e. when false is returned), we can log the message in the base class?

**Comment by Rishabh-V** [comment]
After you address these, please request review from Jon/Amanda. Thanks.

**Comment by anuragsrivstv** [comment]
Hi @jskeet , @amanda-tarafa Can you please have a look at this PR. Thanks.

**Review by amanda-tarafa — APPROVED** [review]
This looks good to me. Just a small suggestion from my side.

Please wait for @jskeet to review as well.
And confirm that integration tests are passing locally both if the DB exist and if it doesn't.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs
Comment (jskeet) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs:127]: Createa => Creates

**Review by jskeet — APPROVED** [review]
LGTM with Amanda's suggestion and "please test run integration tests" part as well.

**Comment by anuragsrivstv** [comment]
Currently Re-running all integration tests for success and failure cases of database creation.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabase.cs
Comment (Rishabh-V) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabase.cs:80]:
```suggestion
        catch (SpannerException e) when (e.RpcException?.StatusCode == StatusCode.AlreadyExists)
```

**Review by jskeet — APPROVED** [review]
Yup, this looks fine to me. Please rebase/squash to a single commit before merging.

**Comment by anuragsrivstv** [comment]
Merging now

**Pull request merged**
